core/src/Kokkos_Tuners.hpp

```diff
@@ -533,6 +535,115 @@ class TeamSizeTuner : public ExtendableTunerMixin<TeamSizeTuner> {
         policy.impl_set_vector_length(vector_length);
       }
     }
+    return policy;
+  }
+  void end() {
+    if (Kokkos::Tools::Experimental::have_tuning_tool()) {
+      tuner.end();
+    }
+  }
+
+  TunerType get_tuner() const { return tuner; }
+};
+namespace Impl {
```

Comment:
Utilities to allow us to know that type `double` is associated with kokkos_value_double, and retrieve the `double` from a Kokkos VariableValue